providers/aws/connection/awsec2ebsconn/destroy.go

```diff
@@ -15,6 +15,9 @@ import (
 )
 
 func (c *AwsEbsConnection) DetachVolumeFromInstance(ctx context.Context, volume *awsec2ebstypes.VolumeInfo) error {
+	if volume == nil {
+		return nil // no-op
```

Comment:
fixes second panic where we try to access the `volume.Id` when this is `nil`.

Reply:
I wonder why we are calling detach when we couldn't attach the volume.

At least, I understood that we also couldn't attach it. So, perhaps the fix might be to not call this?